### DISCUSSION TITLE
Stucked by the lack of filter on Directus Users.role

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#16736] [Feature Requests]
Context:

- My regular users should be able to choose between different roles.
- They also have to see Administrators (to be able to send them comments).

Attempts:
- If I restrict the Read of roles to only non Administrator, the regular users will get an error when clicking on an Administrator profile (as the role field cannot be read).

- If I set the role not to be seen, they will not be able to change their own role.

- If I allow the Administrator Role to be Read, they will be able to elevate their own role to an administrator role.

As a work around, I thought to have to duplicate the role field to a fake role (O2O) on which I can put a filter to let them change their own role, hiding the role. But even when the users role is marked unRead, it is still shown in the profile and therefore throws an error.

Allowing to filter Directus Users.role in the Directus Users model would solve the situation.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
